master/pkg/stream/stream_test.go

```diff
@@ -90,31 +124,262 @@ func TestConfigureSubscription(t *testing.T) {
+	event = Event[TestMsgTypeA]{After: &afterMsg}
+	publisherTwo.Broadcast([]Event[TestMsgTypeA]{event})
+
+	require.Equal(t, 1, len(streamer.Msgs), "upsert message was not upserted")
+	upsertMsg, ok := streamer.Msgs[0].(UpsertMsg)
+	require.True(t, ok, "message was not an upsert type")
+	require.Equal(t, 3, int(upsertMsg.Msg.SeqNum()), "Sequence number incorrect")
+
+	beforeMsg = TestMsgTypeA{
+		Seq: 4,
+		ID:  4,
+	}
+	event = Event[TestMsgTypeA]{Before: &beforeMsg}
+	publisherTwo.Broadcast([]Event[TestMsgTypeA]{event})
+
+	require.Equal(t, 2, len(streamer.Msgs), "upsert message was not upserted")
+	deleteMsg, ok := streamer.Msgs[1].(DeleteMsg)
+	require.True(t, ok, "message was not a delete type")
+	require.Equal(t, "4", deleteMsg.Deleted, "Deleted number incorrect")
+
+	// Msgs on publisher should not be sent.
+	afterMsg = TestMsgTypeA{
+		Seq: 3,
+		ID:  3,
+	}
+	event = Event[TestMsgTypeA]{After: &afterMsg}
+	publisher.Broadcast([]Event[TestMsgTypeA]{event})
+
+	require.Equal(t, 2, len(streamer.Msgs), "upsert message was not upserted")
+}
+
+func TestBroadcastSeparateEvents(t *testing.T) {
+	streamer := NewStreamer(prepareNothing)
+	streamerTwo := NewStreamer(prepareNothing)
+	publisher := NewPublisher[TestMsgTypeA]()
+	publisherTwo := NewPublisher[TestMsgTypeB]()
+	publisherThree := NewPublisher[TestMsgTypeB]()
+	trueSub := NewSubscription[TestMsgTypeA](streamer, publisher, alwaysTrue[TestMsgTypeA], alwaysTrue[TestMsgTypeA])
+	separateSub := NewSubscription[TestMsgTypeB](
+		streamerTwo, publisherTwo, alwaysTrue[TestMsgTypeB], alwaysTrue[TestMsgTypeB])
+	togetherSub := NewSubscription[TestMsgTypeB](
+		streamer, publisherThree, alwaysTrue[TestMsgTypeB], alwaysTrue[TestMsgTypeB])
+	trueSub.Register()
+	separateSub.Register()
+	togetherSub.Register()
+
+	// Msgs sent on publisher should be picked up.
+	afterMsg := TestMsgTypeA{
+		Seq: 0,
+		ID:  0,
+	}
+	event := Event[TestMsgTypeA]{After: &afterMsg}
+	publisher.Broadcast([]Event[TestMsgTypeA]{event})
+
+	require.Equal(t, 1, len(streamer.Msgs), "picked up message we don't want")
+	upsertMsg, ok := streamer.Msgs[0].(UpsertMsg)
+	require.True(t, ok, "message was not an upsert type")
+	require.Equal(t, 0, int(upsertMsg.Msg.SeqNum()), "Sequence number incorrect")
+
+	beforeMsg := TestMsgTypeA{
+		Seq: 1,
+		ID:  1,
+	}
+	event = Event[TestMsgTypeA]{Before: &beforeMsg}
+	publisher.Broadcast([]Event[TestMsgTypeA]{event})
+	require.Equal(t, 2, len(streamer.Msgs), "picked up message we don't want")
+	deleteMsg, ok := streamer.Msgs[1].(DeleteMsg)
+	require.True(t, ok, "message was not a delete type")
+	require.Equal(t, "1", deleteMsg.Deleted, "Deleted number incorrect")
+
+	// Msgs sent on publisherTwo should not be picked up.
+	afterMsgJr := TestMsgTypeB{
+		Seq: 2,
+		ID:  2,
+	}
+	eventJr := Event[TestMsgTypeB]{After: &afterMsgJr}
+	publisherTwo.Broadcast([]Event[TestMsgTypeB]{eventJr})
+
+	require.Equal(t, 2, len(streamer.Msgs), "picked up message we don't want")
+
+	beforeMsgJr := TestMsgTypeB{
```

Comment:
it still looks like you've got some Jr's left behind haha

Reply:
should be fixed!